### ISSUE TITLE
[Question] Document Size and Performance

### ISSUE THREAD

**Issue by iamrecursion** [victorb/obsidian-wielder#4]
I'd love to try this plugin out, but I'm slightly worried that due to the eager evaluation model (it seems to try and parse all code blocks in a page eagerly) it won't cope with large files? I don't necessarily want to use it in massive files, but I'm more worried about opening a huge file in general and having things bog down. Are there any plans to make it less eagerly evaluated?

**Comment by victorb** [comment]
Yeah, https://github.com/victorb/obsidian-wielder/issues/2 would alleviate this concern a bit I believe. Another way could be to not do anything with any code block until the user presses <shortcut> or a "Run" button next to the code block.

Another thing you could do already is changing the [Code-block Language](https://wielder.victor.earth/Settings/Code-block+Language) setting to something else than `clojure` (like `clojure-eval`), and then Wielder would only look for code-blocks defined with `clojure-eval` and evaluate those instead, so if you have a large document with tons of `clojure` blocks already, this could help you.

Personally, I tend to split my documents over many pages once they get over a certain size (because it becomes easier to manage and less jumping up/down to find stuff), so I haven't hit this issue myself yet.

But if it's a issue for you, we definitely should try something that could help prevent any performance issues. You think the above solutions could work in your case?

**Comment by iamrecursion** [comment]
Ooh! The code block language setting sounds perfect, but my install doesn't seem to have any settings at the moment? Is there something I have to do specially to add the settings pane?

**Comment by victorb** [comment]
Sorry for the delay @iamrecursion, somehow didn't get notified of your comment...

The settings panel should appear somewhere in the bottom of the general settings panel for Obsidian.

Looks like this for me:

![image](https://user-images.githubusercontent.com/459764/176257728-6695b597-8560-45f9-84e6-cf7498d167df.png)

**Comment by iamrecursion** [comment]
Strange. I'm not seeing it. 

<img width="329" alt="image" src="https://user-images.githubusercontent.com/5780639/176284671-a3b57f1f-0b75-4de8-ba06-5fd15ef1e9b7.png">

I wonder if it's related to the first error in #5?
